packages/x-scheduler/src/joy/event-calendar/store.ts

```diff
@@ -54,21 +54,45 @@ export const selectors = {
     (state: State) => state.visibleResources,
     (events, visibleResources) => {
       const map = new Map<string, CalendarEvent[]>();
+      const allDayEventsMap = new Map<string, CalendarEvent[]>();
       for (const event of events) {
         if (event.resource && visibleResources.get(event.resource) === false) {
           continue; // Skip events for hidden resources
         }
-
         const dayKey = adapter.format(event.start, 'keyboardDate');
+        if (event.allDay) {
+          if (!allDayEventsMap.has(dayKey)) {
+            allDayEventsMap.set(dayKey, []);
+          }
+          allDayEventsMap.get(dayKey)!.push(event);
+          continue; // Only consider all-day events for this map
+        }
+
         if (!map.has(dayKey)) {
           map.set(dayKey, []);
         }
         map.get(dayKey)!.push(event);
       }
 
-      return (day: SchedulerValidDate) => {
+      return (day: SchedulerValidDate, startDateOfView?: SchedulerValidDate) => {
         const dayKey = adapter.format(day, 'keyboardDate');
-        return map.get(dayKey) || [];
+        let allDayEvents;
+        if (startDateOfView && adapter.isSameDay(day, startDateOfView)) {
+          allDayEvents = events.filter(
+            (event) =>
+              event.allDay &&
+              (!event.resource || !visibleResources.get(event.resource)) &&
+              adapter.startOfDay(event.start) <= adapter.startOfDay(day) &&
+              adapter.startOfDay(event.end) >= adapter.startOfDay(day),
+          );
```

Comment:
I'm trying to cover an edge case here. If we are on a week view for instance, and have events that started last week, but end this week, they should be rendered starting from the first column (first displayed day of the week) 
Super open to suggestions on this one

<img width="1047" height="390" alt="image" src="https://github.com/user-attachments/assets/6233aa87-8323-4920-b941-88a007a7304b" />
<img width="1583" height="280" alt="image" src="https://github.com/user-attachments/assets/e8210396-d1b7-43ae-b917-44b84a53730a" />

Reply:
I think this selector is becoming too crowded 😆 
I tried reworking it so that it returns the list of events that should be rendered in a given day.
Feel free to modify it if it doesn't suit your needs.

To handle all day events on the month view, you will have to set `shouldOnlyRenderEventInOneCell` to `true` and handle the spanning correctly.

On the agenda view it now renders the events in all of its days.

And we can clean the function that splits all day and non all day events (based on yesterday's discussion)  depending on how we want to handle non-all-day but multi-days events.
For now I kept it in the `DayTimeGrid.tsx` file for simplicity.